resources/qml/TimelineRow.qml

```diff
@@ -71,27 +71,51 @@ Item {
         gesturePolicy: TapHandler.ReleaseWithinBounds
     }
 
-    RowLayout {
+    Item {
         id: row
 
         anchors.rightMargin: 1
-        anchors.leftMargin: Nheko.avatarSize + 16
+        anchors.leftMargin: Nheko.avatarSize + 12
         anchors.left: parent.left
         anchors.right: parent.right
+        height: msg.height+(reactionRow.height> 0 ? reactionRow.height-4 : 0)
+        Rectangle {
+            anchors.fill: msg
+            property color userColor: TimelineManager.userColor(userId, Nheko.colors.base)
+            property color bgColor: Nheko.colors.base
+            color: Qt.rgba(userColor.r*0.1+bgColor.r*0.9,userColor.g*0.1+bgColor.g*0.9,userColor.b*0.1+bgColor.b*0.9,1) //TimelineManager.userColor(userId, Nheko.colors.base)
+            radius: 4
+            visible: Settings.bubbles
+        }
```

Comment:
This would probably be a bit cleaner, if you made this a Component instead of an Item:

```
    Item {
        id: row

        anchors.rightMargin: 1
        anchors.leftMargin: Nheko.avatarSize + 12
        anchors.left: parent.left
        anchors.right: parent.right
        height: msg.height+(reactionRow.height> 0 ? reactionRow.height-4 : 0)

        // If you want the bubble to be bigger, set a padding here

        background: Settings.bubbles ? Rectangle {
            property color userColor: TimelineManager.userColor(userId, Nheko.colors.base)
            property color bgColor: Nheko.colors.base
            color: Qt.rgba(userColor.r*0.1+bgColor.r*0.9,userColor.g*0.1+bgColor.g*0.9,userColor.b*0.1+bgColor.b*0.9,1) //TimelineManager.userColor(userId, Nheko.colors.base)
            radius: 4
        } : null

Reply:
that'll put the bubble around the reactionRow as well. I'd have put a Component around the GridLayout only. But isn't a Control the more suitable choice? If I understand correctly, a Component is intended for reusable code. I could use that for the metadata, if I want to optionally take it out of the GridLayout.

Reply:
Argh, it is too late. I meant to put Cotntrol there... Not Item or Component... You can actually place the reactionsRow inside the Control, if you want. You just need to set the bottomInset to shrink the bottom border of the Bubble.

Reply:
This graphic is pretty useful for that: https://doc.qt.io/qt-5/qml-qtquick-controls2-control.html#control-layout